homeassistant/components/onvif/device_trigger.py

```diff
@@ -0,0 +1,78 @@
+"""Provides device automations for ONVIF."""
+from typing import List, Optional
+
+import voluptuous as vol
+
+from homeassistant.components.automation import AutomationActionType
+from homeassistant.components.device_automation import DEVICE_TRIGGER_BASE_SCHEMA
+from homeassistant.components.homeassistant.triggers import event as event_trigger
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_DEVICE_ID, CONF_DOMAIN, CONF_PLATFORM, CONF_TYPE
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant
+from homeassistant.helpers.typing import ConfigType
+
+from ...helpers.trigger import TriggerInfo
+from .const import CONF_SUBTYPE, CONF_UNIQUE_ID, DOMAIN
+from .device import ONVIFDevice
+
+TRIGGER_SCHEMA = DEVICE_TRIGGER_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_UNIQUE_ID): str,
+        vol.Required(CONF_TYPE): str,
+        vol.Required(CONF_SUBTYPE): str,
+    }
+)
+
+
+async def async_get_triggers(
+    hass: HomeAssistant, device_id: str
+) -> List[dict[str, str]]:
+    """List device triggers for ONVIF devices."""
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device_conf = device_registry.async_get(device_id)
+
+    triggers: list[dict[str, str]] = []
+    for entry_id in device_conf.config_entries:
+        config_entry: Optional[ConfigEntry] = hass.config_entries.async_get_entry(
+            entry_id
+        )
+        assert config_entry is not None
+
+        if config_entry.domain != DOMAIN:
+            continue
+
+        device: ONVIFDevice = hass.data[DOMAIN][config_entry.unique_id]
+        for event in device.events.get_platform("event"):
```

Comment:
As I understand it, the possible onvif events defined in `homeassistant/components/onvif/parsers.py`?
If so, the events are currently using a name which is English natural language.

For the device triggers, it would be better to map non natural language to natural language, for example: `motion_alarm` ->`Motion Alarm`

Maybe add a new key `event_id` to `homeassistant.components.onvif.models.Event`?

Note: Translatable entity names will come soon: <https://github.com/home-assistant/core/pull/88242>, the entities generated from events should then have their names translated.

Reply:
I wasn't sure this was relevant as the event name is only used in the subtype of the event, nonetheless I implemented a conversion to non-natural language.